### DISCUSSION TITLE
More coding styles

### DISCUSSION THREAD

**Discussion by togfoxy** [togfoxy/MarsLander#205] [Ideas]
Thinking we should document up front the practice of renaming native love functions into something that is more personally convenient. Like this.

```
function drawImage(img,x,y)
    love.graphics.draw(img,x,y)
end
```

The function serves takes a native love function and renames it. 

I think it makes it hard to read code and creates a barrier for new devs. If we think we should avoid this then perhaps we can put this in the dev wiki?

**Comment by MadByteDE** [comment]
I think doing things like
```lua
local newImage = love.graphics.newImage

```
is something we have to avoid in the game code everywhere expect for libraries.

Id say we stick to writing out love.graphics.draw instrad of changing it.

We don't have to enforce that ppl have to use a specific style as well to sent jn PRs. If someone creates a PR with something like
```lua
local lg = love.graphics
...

lg.draw(Image, x ,y)

```

We shouldn't reject it, but mention that we likely will reformat it in future or ask thr dev to change it to match the rest of the code but still allowing the PR.

**Comment by togfoxy** [comment]
Okay. Let's put in wiki that renaming native functions without good reason is discouraged but is a legitimate candidate for future refactoring.
